model/common/src/icon4py/model/common/grid/vertical.py

```diff
@@ -147,6 +147,23 @@ def __str__(self):
         vertical_params_properties.extend(array_value)
         return "\n".join(vertical_params_properties)
 
+    def start_index(self, domain: Domain):
```

Comment:
These you don't need. But you got the idea... There is a function `def index(domain)` which will do the resolution for you. Whether it is start or end is determined by whether you use it with the `vertical_start` and `vertical_end` argument.

Reply:
Also I fixed some stuff and pusehd which was wrong for the vertical domains to the the [prototype factory branch](https://github.com/C2SM/icon4py/tree/prototype_fields_factory)